### DISCUSSION TITLE
Bulk-Edit Audiobook Meta Data

### DISCUSSION THREAD

**Discussion by beryl03** [advplyr/audiobookshelf#3154] [Ideas]
**Want:**
I want to edit meta-data (edit tags) on a whole series quickly.

**Problem:**
ABS webpage only seems to allow editing of a single book at a time. It is possible to select multiple books, but each entry shows up in a list. Also, there does not seem to be an edit option on a series, and collapse a series does not seem to help either.

**Comment by nichwall** [comment]
You can select the little circle in the upper left of a book cover (on either the library or series view) to select multiple books and use the "Bulk Edit" in the upper right of the page.

You can also Shift+Click on a book to select all of the books in between that book and the most recently selected book (so you can do several ranges). The interface gets slow if you select too many books at a time.

**Comment by beryl03** [comment]
**shift-click** works to select multiple items. Could be nice to do the selection at a series level.
![image](https://github.com/user-attachments/assets/1470a427-a611-4e5b-8415-84d9a8e0cd8c)
1) is "Edit" (not bulk edit, and shows each book sequentially (it would take a long time for 50-item series or podcasts).
2) is the "hamburger" menu, and the expanded box below it does the available options. I see no "Bulk Edit" button.

My use case is to apply Tags to books in a series. It would be time consuming to apply the same tag to 150 items over 2 hours one at a time.

**Reply by nichwall** [reply]
The orange edit button is bulk edit. The list that pops up is so you can edit individual fields on one book as needed.

![bulk_edit_main](https://github.com/user-attachments/assets/be2e2e3b-747b-4473-8e67-6879e3dc55c8)

If you want to add something to or replace the values in a field, use the "Map" menu at the top of the bulk edit screen (circled in red).
![bulk_edit_map_location](https://github.com/user-attachments/assets/34f70e57-7187-4e78-b9c4-4ee5b052b4bb)

Then, you can choose to "Overwrite" to clear data in specific fields and put new data in if it's supplied, or "Append" new data to the selected fields (circled in blue). If you want to add a new tag to all of the books for the bulk edit, use "Append" and then check the "Tags" field (circled in yellow), then press the Apply button to save the change. If only the "Tags" field is checked, that is the only field that will be changed.
![bulk_edit_tags](https://github.com/user-attachments/assets/22a315df-163c-43ac-8396-62dc3a538ad9)

**Reply by jcizzle209** [reply]
I used this before the last couple of updates, but now it doesnt seem to work.  it wont save the changes.  I've tried deleting the subtitles on multipl books and it wont save anymore.  any ideas?

**Reply by JonasFocke01** [reply]
@nichwall 
I want to add the selected books to a series. While that works fine as you described it, it would be really cool if i could input the series #N of the books in question. Where N is the number of the book in the series. The N could be grabbed from the title of the book.

In my case `My title - 057 - my subtitle` would result in a series of `<custom series> #057`.

Is that possible? If not, is this a desirable feature for audiobookshelf?

**Comment by beryl03** [comment]
TLDR; it works, but could be more intuitive.

Thank you!
These instructions accomplish most of what I wanted.
I do still wish that the method was more intuitive. By that I mean that when selecting multiple items (e.g. books, collections, ...) it would automatically infer that you want to edit ALL of the selected items. I have used other programs, mp3tag being one of them, that already has behaviour like this. If requested, I can expand upon this idea in more detail.

If the developers want to update the UI with this idea, great! if not, ok. I am of the opinion that capability and usability are of equal importance.
